pkg/testexecutionservice/testexecution.go

```diff
@@ -88,51 +88,64 @@ func (tes *testExecutionService) Run(ctx context.Context,
+		var cmd *exec.Cmd
+		if tasConfig.Framework == "jasmine" || tasConfig.Framework == "mocha" {
+			if collectCoverage {
+				cmd = exec.CommandContext(ctx, "nyc", commandArgs...)
+			} else {
+				cmd = exec.CommandContext(ctx, commandArgs[0], commandArgs[1:]...) //nolint:gosec
+			}
 		} else {
-			cmd = exec.CommandContext(ctx, commandArgs[0], commandArgs[1:]...)
+			cmd = exec.CommandContext(ctx, commandArgs[0], commandArgs[1:]...) //nolint:gosec
+			if collectCoverage {
+				envVars = append(envVars, "TAS_COLLECT_COVERAGE=true")
+			}
 		}
-	} else {
-		cmd = exec.CommandContext(ctx, commandArgs[0], commandArgs[1:]...)
-		if collectCoverage {
-			envVars = append(envVars, "TAS_COLLECT_COVERAGE=true")
+		cmd.Dir = global.RepoDir
+		cmd.Env = envVars
+		cmd.Stdout = maskWriter
+		cmd.Stderr = maskWriter
+		tes.logger.Debugf("Executing test execution command: %s", cmd.String())
+		if err := cmd.Start(); err != nil {
+			tes.logger.Errorf("failed to execute test %s %v", cmd.String(), err)
+			return nil, err
 		}
-	}
-	cmd.Dir = global.RepoDir
-	cmd.Env = envVars
-	cmd.Stdout = maskWriter
-	cmd.Stderr = maskWriter
-
-	tes.logger.Debugf("Executing test execution command: %s", cmd.String())
-	if err := cmd.Start(); err != nil {
-		tes.logger.Errorf("failed to execute test %s %v", cmd.String(), err)
-		return nil, err
-	}
-	pid := int32(cmd.Process.Pid)
-	tes.logger.Debugf("execution command started with pid %d", pid)
+		pid := int32(cmd.Process.Pid)
+		tes.logger.Debugf("execution command started with pid %d", pid)
 
-	if err := tes.ts.CaptureTestStats(pid, payload.CollectStats); err != nil {
-		tes.logger.Errorf("failed to find process for command %s with pid %d %v", cmd.String(), pid, err)
-		return nil, err
-	}
-
-	errC := cmd.Wait()
-	if errC != nil {
-		tes.logger.Errorf("Error in executing []: %+v\n", errC)
+		if err := tes.ts.CaptureTestStats(pid, payload.CollectStats); err != nil {
+			tes.logger.Errorf("failed to find process for command %s with pid %d %v", cmd.String(), pid, err)
+			return nil, err
+		}
+		if err := cmd.Wait(); err != nil {
+			tes.logger.Errorf("Error in executing []: %+v\n", err)
+			return nil, err
+		}
+		result := <-tes.ts.ExecutionResultOutputChannel
+		executionResults.Results = append(executionResults.Results, result.Results...)
 	}
-	executionResults := <-tes.ts.ExecutionResultOutputChannel
-
 	// FIXME:  commenting this out as we will need to rework on coverage logic after test parallelization
 	// if collectCoverage {
 	// 	if err := tes.createCoverageManifest(tasConfig, coverageDir, removedfiles, executeAll); err != nil {
 	// 		tes.logger.Errorf("failed to create manifest file %v", err)
 	// 		return nil, err
 	// 	}
 	// }
-	return executionResults, errC
+	azureWriter.Close()
+	if uploadErr := <-errChan; uploadErr != nil {
+		tes.logger.Errorf("failed to upload logs for test execution, error: %v", uploadErr)
+		return nil, uploadErr
+	}
+	return executionResults, nil
```

Comment:
why is this added?

Reply:
to return execution results . We are using this to find task fail or pass

Reply:
Line 144 to 148 we already exist in code..

Reply:
removed these lines. these are added by mistake during conflict resolve process.

Reply:
in case any exceptions in tests will get test failure only.